src/shared/queries.ts

```diff
@@ -26,7 +26,7 @@ const accountFragment = gql`
 
 export const insertAccount = gql`
   mutation($account: auth_accounts_insert_input!) {
-    insert_auth_accounts(objects: [$account]) {
+    insert_auth_accounts(objects: [$account], on_conflict: {constraint: accounts_email_key, update_columns: [email]}) {
```

Comment:
Do we really need to update `email` if if we hit the `accounts_email_key` constraint? Can the array be empty like: `update_columns: []`?

Reply:
I tried it but failed with an error. Let me check

Reply:
Since we are taking the affected rows as a response, if we set it to an empty array, it will not affect any rows. So the response will not contain updated rows.

```
Error: cannot proceed to insert array relations since insert to table "auth.accounts" affects zero rows: {"response":{"errors":[{"extensions":{"path":"$.selectionSet.insert_auth_accounts.args.objects[0]","code":"not-supported"},"message":"cannot proceed to insert array relations since insert to table \"auth.accounts\" affects zero rows"}],"status":200},"request":{"query":"mutation ($account: auth_accounts_insert_input!) {\n  insert_auth_accounts(objects: [$account], on_conflict: {constraint: accounts_email_key, update_columns: []}) {\n    affected_rows\n    returning {\n      ...accountFragment\n    }\n  }\n}\n\nfragment accountFragment on auth_accounts {\n  id\n  active\n  default_role\n  account_roles {\n    role\n  }\n  user {\n    id\n    display_name\n    name\n  }\n  is_anonymous\n  ticket\n  email\n  new_email\n  otp_secret\n  mfa_enabled\n  password_hash\n}\n","variables":{"account":{"email":"...","password_hash":null,"active":true,"default_role":"user","account_roles":{"data":[{"role":"user"}],"on_conflict":{"constraint":"user_roles_account_id_role_key","update_columns":[]}},"user":{"data":{"display_name":"....","avatar_url":"...."}},"account_providers":{"data":[{"auth_provider":"google","auth_provider_unique_id":"...."}]}}}}}
    at GraphQLClient.<anonymous> (/app/node_modules/graphql-request/src/index.ts:72:13)
    at step (/app/node_modules/graphql-request/dist/src/index.js:40:23)
    at Object.next (/app/node_modules/graphql-request/dist/src/index.js:21:53)
    at fulfilled (/app/node_modules/graphql-request/dist/src/index.js:12:58)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  response: { errors: [ [Object] ], status: 200 },
  request: {
    query: 'mutation ($account: auth_accounts_insert_input!) {\n' +
      '  insert_auth_accounts(objects: [$account], on_conflict: {constraint: accounts_email_key, update_columns: []}) {\n' +
      '    affected_rows\n' +
      '    returning {\n' +
      '      ...accountFragment\n' +
      '    }\n' +
      '  }\n' +
      '}\n' +
      '\n' +
      'fragment accountFragment on auth_accounts {\n' +
      '  id\n' +
      '  active\n' +
      '  default_role\n' +
      '  account_roles {\n' +
      '    role\n' +
      '  }\n' +
      '  user {\n' +
      '    id\n' +
      '    display_name\n' +
      '    name\n' +
      '  }\n' +
      '  is_anonymous\n' +
      '  ticket\n' +
      '  email\n' +
      '  new_email\n' +
      '  otp_secret\n' +
      '  mfa_enabled\n' +
      '  password_hash\n' +
      '}\n',
    variables: { account: [Object] }
  }
}
(node:34) UnhandledPromiseRejectionWarning: Error: Internal Server Error
```